rpmio/digest_openssl.c

```diff
@@ -288,56 +288,56 @@ int rpmDigestFinal(DIGEST_CTX ctx, void ** datap, size_t *lenp, int asAscii)
 /* Key */
 
 struct pgpDigKeyRSA_s {
-    size_t nbytes; /* Size of modulus */
-
     BIGNUM *n; /* Common Modulus */
     BIGNUM *e; /* Public Exponent */
-
     EVP_PKEY *evp_pkey; /* Fully constructed key */
+    unsigned int poison: 1; /* if set, this key cannot be mutated */
```

Comment:
Defaulting to 1 but relying in the code to calloc it to 0 is a bit confusing - even if the code seems fine.

Reply:
Arg, I had something ringing in the back of my mind that I didn't get this quite right. Thanks for clarifying.